File: tcp.py
import socket

def handle_client(client_socket):
    client_socket.sendall(b"Open")
    while True:
        data = client_socket.recv(1024)
        if not data:
            break
        print(data.decode())

    client_socket.close()

server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
server.bind(('0.0.0.0', 8080))
server.listen(5)

while True:
    client_socket, addr = server.accept()
    handle_client(client_socket)

File: tcp.js
const net = require("net")


const server = net.createServer(socket => {
    socket.write("Open")
    socket.on("data",data => {
        console.log(data.toString())
        socket.write("Received your data: " + data.toString());
    })
    socket.on("end", () => {
        console.log("Client disconnected");
    });
})

console.log('server running')
server.listen(8080)